Update library overview [8399] (#118)
eProsima/Fast-RTPS-docs: docs/fastdds/library_overview/includes/exec_model.rst

methods on the same DomainParticipant from different threads.
However, this multithreading implementation must be taken into account when external functions access to resources that
are modified by threads running internally in the library.
An example of this is the modified resources in the entity listener callbacks.
The following is a brief overview of how Fast DDS multithreading schedule work:

* Main thread: Managed by the application.
* Event thread: Each DomainParticipant owns one of these. It processes periodic and triggered time events.
* Asynchronous writer thread: This thread manages asynchronous writes for all DomainParticipants.
  Even for synchronous writers, some forms of communication must be initiated in the background.
* Reception threads: DomainParticipants spawn a thread for each reception channel, where the concept of a channel
  depends on the transport layer (e.g. a UDP port).

Event-driven architecture
^^^^^^^^^^^^^^^^^^^^^^^^^

There is a time-event system that enables Fast DDS to respond to certain conditions, as well as schedule periodic
operations.

> Few of them are visible to the user since most are related to DDS and RTPS metadata.

Reviewer: ```suggestion
Few of them are visible to the user since they are mostly related to DDS and RTPS metadata.
```

> However, the user can define in his/her application periodic time-events by inheriting from the :class:`TimedEvent`

Reviewer: ```suggestion
However, the user can define periodic time-events in their application by inheriting from the :class:`TimedEvent`
```
